Client/Frontend/Settings/Brave Today/BraveTodayAddSourceViewController.swift

```diff
@@ -0,0 +1,390 @@
+            }
+        }
+        pageTask?.resume()
+    }
+    
+    private let textField = UITextField().then {
+        $0.attributedPlaceholder = NSAttributedString(
+            string: Strings.BraveToday.searchTextFieldPlaceholder,
+            attributes: [.foregroundColor: UIColor.lightGray]
+        )
+        $0.font = .preferredFont(forTextStyle: .body)
+        $0.keyboardType = .URL
+        $0.autocorrectionType = .no
+        $0.autocapitalizationType = .none
+        $0.returnKeyType = .search
+    }
+    
+    private var isSearchEnabled: Bool {
+        if let text = textField.text {
+            return URIFixup.getURL(text) != nil
+        }
+        return false
+    }
+    
+    // MARK: - UITableViewDelegate
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        if indexPath.section == 0 && indexPath.row == 1, isSearchEnabled, !isLoading {
+            searchPageForFeeds()
+        }
+        if indexPath.section == 1 && indexPath.row == 0 {
+            tappedImportOPML()
+        }
+        tableView.deselectRow(at: indexPath, animated: true)
+    }
+    
+    override func tableView(_ tableView: UITableView, shouldHighlightRowAt indexPath: IndexPath) -> Bool {
+        if indexPath.section == 0 {
+            if indexPath.row == 1 {
+                return isSearchEnabled && !isLoading
+            }
+            return false
+        }
+        return true
+    }
+    
+    // MARK: - UITableViewDataSource
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        switch indexPath.section {
+        case 0:
+            switch indexPath.row {
+            case 0:
+                let cell = tableView.dequeueReusableCell(for: indexPath) as FeedSearchCellClass
+                cell.textField = textField
+                return cell
+            case 1:
+                let cell = tableView.dequeueReusableCell(for: indexPath) as CenteredButtonCell
+                cell.textLabel?.text = Strings.BraveToday.searchButtonTitle
+                cell.tintColor = isSearchEnabled && !isLoading ? BraveUX.braveOrange : Colors.grey500
+                return cell
+            default:
+                fatalError("No cell available for index path: \(indexPath)")
+            }
+        case 1:
+            let cell = tableView.dequeueReusableCell(for: indexPath) as CenteredButtonCell
+            cell.textLabel?.text = Strings.BraveToday.importOPML
+            cell.tintColor = BraveUX.braveOrange
+            return cell
+        default:
+            fatalError("No cell available for index path: \(indexPath)")
+        }
+    }
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        2
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+        case 0: return 2
```

Comment:
Maybe (optional) we use an enum for the sections?

Reply:
Thought about it, but for only 2 sections which essentially display the same data I opted to just leave it